src/amo/components/Addon/index.js

```diff
@@ -73,9 +80,13 @@ export class AddonBase extends React.Component {
   }
 
   componentWillMount() {
```

Comment:
I notice there's nothing here for `componentDidUpdate`–if an add-on linked to another add-on the components on the page wouldn't change but the props would update and we would want to do things like update the `viewContext`.

Of course, it would be nice if we did that more often with routing–as in this case we could get away with using the URL to know `addonType` but for now we should `dispatch(setViewContext())` on `componentDidUpdate`.